src/pages/resultsView/coExpression/CoExpressionViz.tsx

```diff
@@ -0,0 +1,293 @@
+import * as React from "react";
+import {Gene, MolecularProfile, Mutation, NumericGeneMolecularData} from "../../../shared/api/generated/CBioPortalAPI";
+import {observer, Observer} from "mobx-react";
+import CoExpressionTable from "./CoExpressionTable";
+import {action, autorun, computed, IReactionDisposer, observable} from "mobx";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import {
+    SimpleGetterMobXApplicationDataStore,
+    SimpleMobXApplicationDataStore
+} from "../../../shared/lib/IMobXApplicationDataStore";
+import {CoExpression} from "../../../shared/api/generated/CBioPortalAPIInternal";
+import GeneMolecularDataCache from "../../../shared/cache/GeneMolecularDataCache";
+import CoExpressionPlot, {ICoExpressionPlotProps} from "./CoExpressionPlot";
+import {remoteData} from "../../../shared/api/remoteData";
+import MutationDataCache from "../../../shared/cache/MutationDataCache";
+import {cached, MobxPromise} from "mobxpromise";
+import {computePlotData, requestAllDataMessage} from "./CoExpressionVizUtils";
+import {Button} from "react-bootstrap";
+import {MutationCache, NumericGeneMolecularDataCache} from "../ResultsViewPageStore";
+import {CoExpressionCache} from "./CoExpressionTab";
+
+export interface ICoExpressionVizProps {
+    gene:Gene;
+    molecularProfile:MolecularProfile;
+    onClickRequestAllData:()=>void;
+    coExpressionCache:CoExpressionCache;
+    allDataRequested:boolean;
+    numericGeneMolecularDataCache:NumericGeneMolecularDataCache;
+    mutationCache?:MutationCache;
+    setHighlightedCoExpression:(c:CoExpression)=>void;
+    highlightedCoExpression?:CoExpression;
+}
+
+export enum TableMode {
+    SHOW_ALL, SHOW_POSITIVE, SHOW_NEGATIVE
+}
+
+export class CoExpressionDataStore extends SimpleGetterMobXApplicationDataStore<CoExpression> {
```

Comment:
change name of super class so that it indicates that this is LazyMobxTable specific